src/cratedb/mod.rs

```diff
@@ -0,0 +1,133 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use testcontainers::{core::WaitFor, CopyToContainer, Image};
+
+const NAME: &str = "crate";
+const TAG: &str = "5.8";
+
+/// Represents a CrateDB docker instance.
+/// It is based on the official [`crate docker image`]
+///
+/// Defaults are not modified:
+/// super_user: crate
+/// password: <empty>
+/// pg_port: 5432
+/// http_port: 4200
+///
+/// You can connect run instructions by sending an HTTP call to /_sql in port 4200.
+/// [`crate http interface`]
+///
+/// or using the postgres wire compatible protocol at port 5432.
+///
+/// # Example of postgres wire.
+/// use testcontainers_modules::{postgres, testcontainers::runners::SyncRunner};
+///
+/// let postgres_instance = cratedb::CrateDB::default().start().unwrap();
+///
+/// let connection_string = format!(
+///     "postgres://crate@{}:{}/postgres",
+///     postgres_instance.get_host().unwrap(),
+///     postgres_instance.get_host_port_ipv4(5432).unwrap()
+/// );
+///
+/// let mut conn = postgres::Client::connect(connection_string, postgres::NoTls).unwrap();
+/// let rows = conn.query("SELECT 1 + 1", &[]).unwrap();
+///
+/// [`crate docker image`]: https://hub.docker.com/_/crate
+/// [`crate http interface`]: https://cratedb.com/docs/crate/reference/en/latest/interfaces/http.html
+#[derive(Debug, Clone)]
+pub struct CrateDB {
+    env_vars: HashMap<String, String>,
+    copy_to_sources: Vec<CopyToContainer>,
+}
+
+impl CrateDB {
+    /// Sets CrateDB's heap size, only increase it if you are testing big high volumes of data.
+    pub fn with_heap_size(mut self, ram_gb: usize) -> Self {
+        self.env_vars
+            .insert("CRATE_HEAP_SIZE".to_string(), format!("{}g", ram_gb));
+        self
+    }
+}
+impl Default for CrateDB {
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+        env_vars.insert("CRATE_HEAP_SIZE".to_string(), "1g".to_string());
+        Self {
+            env_vars,
+            copy_to_sources: Vec::new(),
+        }
+    }
+}
+
+impl Image for CrateDB {
+    fn name(&self) -> &str {
+        NAME
+    }
+
+    fn tag(&self) -> &str {
+        TAG
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::message_on_stdout("started")]
+    }
+
+    fn env_vars(
+        &self,
+    ) -> impl IntoIterator<Item = (impl Into<Cow<'_, str>>, impl Into<Cow<'_, str>>)> {
+        &self.env_vars
+    }
+
+    fn copy_to_sources(&self) -> impl IntoIterator<Item = &CopyToContainer> {
+        &self.copy_to_sources
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use testcontainers::{runners::SyncRunner, ImageExt};
+    #[test]
```

Comment:
This fixes the linting issue raised by rustfmt

```suggestion
    use super::*;
    use crate::testcontainers::{runners::SyncRunner, ImageExt};

    #[test]
```